code/modules/mining/aux_base_camera.dm

```diff
@@ -0,0 +1,266 @@
+	desc = "An industrial computer integrated with a camera-assisted rapid construction device."
+	networks = list("SS13")
+	var/obj/item/weapon/rcd/internal/RCD //Internal RCD. The computer passes user commands to this in order to avoid massive copypaste.
+	circuit = /obj/item/weapon/circuitboard/computer/base_construction
+	off_action = new/datum/action/innate/camera_off/base_construction
+	var/datum/action/innate/aux_base/switch_mode/switch_mode_action = new //Action for switching the RCD's build modes
+	var/datum/action/innate/aux_base/build/build_action = new //Action for using the RCD
+	var/datum/action/innate/aux_base/airlock_type/airlock_mode_action = new //Action for setting the airlock type
+	var/datum/action/innate/aux_base/window_type/window_action = new //Action for setting the window type
+	var/datum/action/innate/aux_base/place_fan/fan_action = new //Action for spawning fans
+	var/fans_remaining = 0 //Number of fans in stock.
+	var/datum/action/innate/aux_base/install_turret/turret_action = new //Action for spawning turrets
+	var/turret_stock = 0 //Turrets in stock
+	var/obj/machinery/computer/shuttle/auxillary_base/found_aux_console //Tracker for the Aux base console, so the eye can always find it.
+
+	icon_screen = "mining"
+	icon_keyboard = "rd_key"
+
+/obj/machinery/computer/camera_advanced/base_construction/New()
+	..()
+	RCD = new /obj/item/weapon/rcd/internal(src)
+
+/obj/machinery/computer/camera_advanced/base_construction/Initialize(mapload)
+	..()
+	if(mapload) //Map spawned consoles have a filled RCD and stocked special structures
+		RCD.matter = RCD.max_matter
+		fans_remaining = 4
+		turret_stock = 4
+
+/obj/machinery/computer/camera_advanced/base_construction/CreateEye()
+
+	var/spawn_spot
+	if(!found_aux_console)
+		found_aux_console = locate(/obj/machinery/computer/shuttle/auxillary_base) in machines
+
+		if(found_aux_console)
+			spawn_spot = found_aux_console
+	else
+		spawn_spot = src
+
+
+	eyeobj = new /mob/camera/aiEye/remote/base_construction(get_turf(spawn_spot))
+	eyeobj.origin = src
+
+
+/obj/machinery/computer/camera_advanced/base_construction/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/rcd_ammo) || istype(W, /obj/item/stack/sheet))
+		RCD.attackby(W, user, params) //If trying to feed the console more materials, pass it along to the RCD.
+	else
+		return ..()
+
+/obj/machinery/computer/camera_advanced/base_construction/GrantActions(mob/living/user)
+	off_action.target = user
+	off_action.Grant(user)
+	switch_mode_action.target = src
+	switch_mode_action.Grant(user)
+	build_action.target = src
+	build_action.Grant(user)
+	airlock_mode_action.target = src
+	airlock_mode_action.Grant(user)
+	window_action.target = src
+	window_action.Grant(user)
+	fan_action.target = src
+	fan_action.Grant(user)
+	turret_action.target = src
+	turret_action.Grant(user)
+	eyeobj.invisibility = 0 //When the eye is in use, make it visible to players so they know when someone is building.
```

Comment:
If you're doing this, give it IC sprite. Construction crane or something similar.